conjur/api/ssl_utils/ssl_context_factory.py

```diff
@@ -0,0 +1,81 @@
+# -*- coding: utf-8 -*-
+
+"""
+SslContextFactory module
+This module job is to encapsulate the creation of SSLContext in the dependent of each os environment
+"""
+
+import logging
+import ssl
+from subprocess import Popen, PIPE
+import platform
+import os
+
+from conjur.util.os_types import OSTypes
+from conjur.util.util_functions import get_current_os
+
+
+# pylint: disable=too-few-public-methods
+class SslContextFactory:
+    """
+    Factory to create SSLContext object
+    """
+
+    @classmethod
+    def create_platform_specific_ssl_context(cls) -> ssl.SSLContext:
+        """
+        Factory method to create SSLContext loaded with system RootCA's
+        @return: SSLContext configured with the system certificates
+        """
+        ctx = ssl.create_default_context()
+        if len(ctx.get_ca_certs()) > 0:
+            return ctx
+        current_platform = get_current_os()
+        if current_platform == OSTypes.MAC_OS:
+            cls._configure_ctx_for_mac(ctx)
+        elif current_platform == OSTypes.LINUX:
+            cls._configure_ctx_for_linux(ctx)
```

Comment:
I don't think this is needed.
The badssl tests pass, so get_ca_certs() shouldn't be empty.